thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/AnomalyFlattenResource.java

```diff
@@ -0,0 +1,143 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.thirdeye.dashboard.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import io.swagger.annotations.ApiOperation;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.thirdeye.anomalydetection.context.AnomalyFeedback;
+import org.apache.pinot.thirdeye.common.dimension.DimensionMap;
+import org.apache.pinot.thirdeye.datalayer.bao.MergedAnomalyResultManager;
+import org.apache.pinot.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import org.apache.pinot.thirdeye.datasource.DAORegistry;
+
+
+/**
+ * Flatten the anomaly results for UI purpose.
+ * Convert a list of anomalies to rows of columns so that UI can directly convert the anomalies to table
+ */
+@Path("thirdeye/table")
+public class AnomalyFlattenResource {
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+  private MergedAnomalyResultManager mergedAnomalyResultDAO;
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  public static final String ANOMALY_ID = "anomalyId";
+  public static final String ANOMALY_COMMENT = "comment";
+  public static final String FUNCTION_ID = "functionId";
+  public static final String WINDOW_START = "start";
+  public static final String WINDOW_END = "end";
+  private static final String DIMENSION_KEYS  = "keys";
+
+  private static final List<String> REQUIRED_JSON_KEYS =Arrays.asList(FUNCTION_ID, WINDOW_START, WINDOW_END);
+  private static final String DEFAULT_DELIMINATOR = ",";
+
+  public AnomalyFlattenResource() {
+    this.mergedAnomalyResultDAO = DAO_REGISTRY.getMergedAnomalyResultDAO();
+  }
+
+  public AnomalyFlattenResource(MergedAnomalyResultManager mergedAnomalyResultDAO) {
+    this.mergedAnomalyResultDAO = mergedAnomalyResultDAO;
+  }
+
+  /**
+   * Flatten a list of anomaly results to a list of map
+   * @param jsonPayload a json string; the jsonPayload should include keys: [functionId, start, end].
+   *                    start and end are in the format of epoch time
+   * @return a list of map
+   * @throws IOException
+   */
+  @GET
+  @ApiOperation(value = "View a flatted merged anomalies for collection")
+  public List<Map<String, String>> flatAnomalyResults(String jsonPayload) throws IOException {
```

Comment:
"Get" method can't take any JSON payload so this won't work. 

IMO, the parameters for the endpoint should be explicit. javax.ws.rs should handle all serialization/deserialization.